components/scream/src/physics/p3/p3_functions_f90.cpp

```diff
@@ -2863,7 +2863,7 @@ void check_values_f(Real* qv, Real* temp, Int kstart, Int kend,
 
   kstart -= 1;
   kend -= 1;
-  const Int nk = (kend - kstart) + 1;
+  const unsigned long nk = (unsigned long)((kend - kstart) + 1);
```

Comment:
We should probably settle on Int vs size_t for our indices at some point. Personally I like `int` for indices cause if `m>i`, stuff like `i = i - m` will not wrap around to a very large positive int, but generate a negative number which we can check for in debug mode. On the other hand, size_t is appealing for sizes (which is what is needed here).

Reply:
Yeah, I agree. Where is it appropriate to raise these kinds of issues? Should we create a GitHub issue to track this?

Reply:
Not sure. It's a guideline we should eventually agree on, and put somewhere (both in the repo as on confluence), more than an issue.

A good issue (and good first issue for new comers) could be to implement a `scream_debug_check` (I think `scream_require_msg` runs in both debug and release, while we might want one for debug only), and put it in a ton of places, to avoid leaving inputs unchecked.